src/generateSdk/astGenerator/TsAstGenerator.ts

```diff
@@ -1,166 +1,262 @@
-              type: AstNodeType.NativeType,
-              paramType: this.mapTypesToParamType(method.type)
-            };
-            returnType = isTypeAlias
-              ? (returnTypeAlias as TypeAlias)
-              : returnNativeType;
-          } else {
-            const returnAny: NativeType = {
-              type: AstNodeType.NativeType,
-              paramType: NativeTypeEnum.any
-            };
-            returnType = returnAny;
+  parseTypeAliasDeclaration(typeAliasDeclaration: typescript.Node): StructLiteral | TypeAlias {
+    const typeAliasDeclarationCopy: any = { ...typeAliasDeclaration };
+    delete typeAliasDeclarationCopy.parent;
+    delete typeAliasDeclarationCopy.name.parent;
+    if (typeAliasDeclarationCopy.modifiers) {
+      delete typeAliasDeclarationCopy.modifiers.parent;
+      for (const modifier of typeAliasDeclarationCopy.modifiers) {
+        delete modifier.parent;
+      }
+    }
+    if (typeAliasDeclarationCopy.type) {
+      delete typeAliasDeclarationCopy.type.parent;
+    }
+    if (typeAliasDeclarationCopy.type.kind === typescript.SyntaxKind.TypeLiteral) {
+      const structLiteral: StructLiteral = {
+        type: AstNodeType.StructLiteral,
+        name: typeAliasDeclarationCopy.name.escapedText,
+        typeLiteral: {
+          type: AstNodeType.TypeLiteral,
+          properties: []
+        }
+      }
+      for (const member of typeAliasDeclarationCopy.type.members) {
+        delete member.parent;
+        delete member.name.parent;
+        if (member.type) {
+          delete member.type.parent;
+          const field: PropertyDefinition = {
+            name: member.name.escapedText,
+            type: this.mapTypesToParamType(member.type),
           }
-
-          classDefinition.methods.push({
-            type: AstNodeType.MethodDefinition,
-            name: method.name,
-            static: false,
-            kind: MethodKindEnum.method,
-            returnType: returnType,
-            params: parameters
-          });
+          structLiteral.typeLiteral.properties.push(field);
         }
       }
+      return structLiteral;
+    } else {
+      return {
+        type: AstNodeType.TypeAlias,
+        name: typeAliasDeclarationCopy.name.escapedText,
+        aliasType: this.mapTypesToParamType(typeAliasDeclarationCopy.type)
+      }
     }
+  }
 
-    if (classDefinition) classDefinition.typeDefinitions = [...types];
+  async generateAst(input: AstGeneratorInput): Promise<AstGeneratorOutput> {
+    const fileData = input.file.data;
 
-    if (classDefinition === undefined) {
-      throw new Error("No class definition found");
-    } else {
-      return {
-        program: {
-          body: [classDefinition],
-          originalLanguage: "ts",
-          sourceType: SourceType.module
+    const node = typescript.createSourceFile('test.ts', fileData.toString(), typescript.ScriptTarget.ES2015, true, typescript.ScriptKind.TS);
```

Comment:
what is this test.ts?

Reply:
`createSourceFile` requires a file name as first parameter. the file does not exist and it doesn't create it. i guess it's irelevant what we pass there, but we have to pass a string. should i change it?